src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1606,12 +1607,11 @@ CHIP_ERROR GroupDataProviderImpl::SetKeySet(chip::FabricIndex fabric_index, cons
     for (size_t i = 0; i < in_keyset.num_keys_used; ++i)
     {
         ByteSpan epoch_key(in_keyset.epoch_keys[i].key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
-        uint8_t key[Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
-        MutableByteSpan key_span(key, sizeof(key));
+        MutableByteSpan key_span(keyset.operational_keys[i].value, sizeof(keyset.operational_keys[i].value));
         ReturnErrorOnFailure(Crypto::DeriveGroupOperationalKey(epoch_key, key_span));
-        ReturnErrorOnFailure(Crypto::DeriveGroupSessionId(ByteSpan(key, sizeof(key)), keyset.operational_keys[i].hash));
+        ReturnErrorOnFailure(Crypto::DeriveGroupSessionId(key_span, keyset.operational_keys[i].hash));
+        VerifyOrReturnError(memcmp(kZeroKey, key_span.data(), sizeof(kZeroKey)), CHIP_ERROR_INTERNAL);
```

Comment:
I ... don't understand what this is doing.  `kZeroKey` is an array of `uint8_t *`.  Is it meant to be that?  Are we trying to make sure that `key_span.data()` is not all zeros?

Reply:
kZeroKey is intended to be an array of uint8_t. The superfluous pointer is a typo. Fixed now.